### DISCUSSION TITLE
Need helping accessing config files to extend from on private github repository

### DISCUSSION THREAD

**Discussion by Marc-Sev** [renovatebot/renovate#30783] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hello !

In an github Team with many repositories, we would want to have all the common parts of our configuration files in a dedicated repository, but I do not seem not to be able to make renovate get it.

```
"extends": [
    "local>xxxORGxxx/renovate-config:xxFILExx" #I tried with both local and github
  ],
  "hostRules": [
    {
      "hostType": "api.github.com",
      "matchHost": "api.github.com",
      "token": "{{ secrets.GITHUB_TOKEN }}"
    }
  ]
```

hostRules were the only way I found to try to give Renovate a token to use, but in the logs it seems not to use it (see logs).

I generated a token for myself to test the path Renovate uses and managed to make it work. Then I came with the hostRules idea to try to give it the token, but unsuccessfully.

Thanks for taking the time to read, and even more if you help !
What am I missing? 

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: GET https://api.github.com/repos/xxxORGxxx/xxxREPOxxx/contents/renovate.json = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=98)
```

</details>

**Comment by rarkins** [comment, marked as answer]
You should never override the api.github.com token completely like you've done as that will break the app. Instead you should install renovate into the shared repo so that the app tokens get automatic access to the shared preset. If all repos are private then you can have a private shared config repo, but if you need to use it for public repos too then a public config repo is required.

**Reply by Marc-Sev** [reply]
Thanks for the quick reply !
I did not think of adding the private project in Renovate, thanks to you.
I concede that overriding the api/github.com was a desperate measure
